app/controllers/tickets_controller.rb

```diff
@@ -164,76 +168,80 @@ def create
     else
+      using_hook = false
       @ticket = Ticket.new(ticket_params)
-      if current_user.nil? && !verify_recaptcha
-        return render 'new'
-      end
     end
 
-    send_notification_email
-
-    respond_to do |format|
-      format.html do
-
-        respond_to_html
-
+    if ::Rails.application.config.ticket_is_open_to_the_world  == false && current_user.nil? && using_hook == false
+      render :status => :forbidden, :text => "Access Denied"
+    elsif can_create_a_ticket(using_hook) && save_current_ticket
+      notify_incoming @ticket
+      
+      respond_to do |format|
+        format.json { render json: @ticket, status: :created }
+        format.html {
+          if current_user.nil?
+            flash.now[:notice] = I18n::translate(:ticket_added_public)
+            render 'create'
+          else
+            redirect_to ticket_url(@ticket), notice: I18n::translate(:ticket_added)
+          end
+        }
       end
-
-      format.json do
-
-        respond_to_json
-
+    else
+      respond_to do |format|
+        format.html {
+          @email_addresses = EmailAddress.verified.ordered
+          render 'new'
+        }
+        format.json {
+          if @ticket.nil?
+            render json: {}, status: :created  # bounce mail handled correctly
+          else
+            render json: @ticket, status: :unprocessable_entity
+          end 
+        }
       end
-
-      format.js { render }
     end
   end
 
   protected
-  def respond_to_json
-    if @ticket.nil?
-      render json: {}, status: :created  # bounce mail handled correctly
-    elsif @ticket.valid?
-      render json: @ticket, status: :created
+  def can_create_a_ticket(using_hook)
+    if @ticket.nil? || !@ticket.valid?
+      flash.now[:alert] = I18n::translate(:form_validation_error)
+      false
+    # relax policy for request coming from emails
+    elsif using_hook
+      true
+    # strict policy for request coming from the application      
     else
-      render json: @ticket, status: :unprocessable_entity
-    end
-  end
-
-  def respond_to_html
-    # not signed in
-    if current_user.nil?
-      # we need to verify the captcha
-      if verify_recaptcha || !Ticket.recaptcha_keys_present?
-        # we need to verify the ticket
-        if !@ticket.nil? && @ticket.valid?
-          render 'create'
-        else
-          # ticket validation failed
-          render_new
+      if Ticket.recaptcha_keys_present? && current_user.nil?
+        if verify_recaptcha
+          true
+        else          
+          flash.now[:alert] = I18n::translate(:captcha_error)
+          false
         end
       else
-        # captcha validation failed
-        render_new
+        true
       end
-    else
-      # signed in we redirect
-      redirect_to ticket_url(@ticket), notice: I18n::translate(:ticket_added)
     end
   end
-
+  def save_current_ticket
+    label = params[:label]
+    if label
+      @ticket.labels.new(:name => label)
```

Comment:
Doesn't this create a new label for every ticket? Instead of re-using the already existing label.